utils/httputils.go

```diff
@@ -17,3 +20,15 @@ func SendPOSTRequest(url string, contentType string, payload []byte) (statusCode
 
 	return resp.StatusCode(), nil
 }
+
+// GetFreeAddrPort returns a loopback address and port that can be listened from.
+// Note: port is included in the address.
+func GetFreeAddrPort() (string, int, error) {
+	l, err := net.Listen("tcp", "localhost:0")
```

Comment:
Cool, is this like a standard way to find a free port?

Reply:
yeap. Re-usable if required